src/middleware/is.psc.validation.middleware.ts

```diff
@@ -0,0 +1,23 @@
+import { NextFunction, Request, Response } from "express";
+import { URL_QUERY_PARAM } from "../types/page.urls";
+import { isPscFlagValid } from "../validators/is.psc.validator";
+import { logger } from "../utils/logger";
+import { Templates } from "../types/template.paths";
+
+export const isPscQueryParameterValidationMiddleware = (req: Request, res: Response, next: NextFunction) => {
+  logger.debug("Execute isPsc query parameter validation middleware checks");
+
+  const isPsc: string = req.query[URL_QUERY_PARAM.IS_PSC] as string;
+
+  if (!isPsc) {
```

Comment:
Do we need this check in two places?

Reply:
the query param will usually not be present and will be undefined. Trouble is we can't filter the url by a query param so all urls will have to enter this middleware and the majority won't have this param and it will be undefined. If we let it through to validation then it will fail (as undefined is not an allowed isPsc value) and the user will be blocked.

Reply:
Worth adding a comment in the code explaining this. I'll do that.